src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -147,6 +147,8 @@ def set_event_handler(event_types: Set[Type[Event]], func: Any) -> None:
                     function = event_handler[type(event)]
                     batch.append((function, event))
                     self._events.task_done()
+                    if type(event) in [EnsembleSucceeded, EnsembleFailed]:
```

Comment:
I made the comment in the issue: https://github.com/equinor/ert/issues/9136#issuecomment-2478709595
This if would make batching slower for no relevant reason.